nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -721,10 +722,30 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
                             try {
                                 DataType targetDataType = DataTypeUtils.getDataTypeFromSQLTypeValue(sqlType);
                                 if (targetDataType != null) {
-                                    currentValue = DataTypeUtils.convertType(
-                                            currentValue,
-                                            targetDataType,
-                                            fieldName);
+                                    if (sqlType == Types.BLOB || sqlType == Types.BINARY) {
+                                        if (currentValue instanceof Object[]) {
```

Comment:
Although the conditional checks for `Object[]` the loop casts each element to a `byte`.  It seems like the check for `Object[]` should be more narrow, and perhaps an exception should be thrown if any other array object type is found?

Reply:
I waffled a bit on that point, but figured if the values could be cast to bytes we'd allow it. Then again, if the field is an Object[Integer], the user might think all their bytes are going in when in fact only 1/4 of them would. I will add an additional check for byte elements